editor/gui/editor_dir_dialog.cpp

```diff
@@ -83,8 +103,7 @@ void EditorDirDialog::config(const Vector<String> &p_paths) {
 }
 
 void EditorDirDialog::reload(const String &p_path) {
-	if (!is_visible()) {
-		must_reload = true;
+	if (!must_reload) {
```

Comment:
This seems strange if it is not visible there is no reasons to reload it. Moreover this might break reload updates if they happen while it is opened (not sure if it can happen).

Reply:
Better to remove this condition and always leave the reload available?

Reply:
No what I meant how the code was structured was when anything it relias on changes it tries to reload imediatly but in cas it is not visible it will mark it as needing to be reload (set `must_reload` to true) and skip it and when made visible it will check if it needs to be reloaded and if it is the case do the reload then. 
The fact is you changed this part of the code and I don't understand why you decided to make that change and what it aims to accomplish. It does not seem like your fix should need any modification to the `must_reload` system that was in place.

Reply:
The dialog won't update unless there's an update in the filesystem dock

https://github.com/godotengine/godot/assets/37383316/ec5d05e5-612a-42ac-b9f5-6ad43f93d2fc

Reply:
then you need `FileSystemDock` to send a signal when colors are modified (might already exists if not you can create one) and connect this signal to the `reload` function. The connection should be done `_notification` under `NOTIFICATION_ENTER_TREE`.